configure.ac

```diff
@@ -255,8 +255,8 @@ AC_ARG_WITH(sandbox-shell, AC_HELP_STRING([--with-sandbox-shell=PATH],
 AC_SUBST(sandbox_shell)
 
 AC_ARG_ENABLE(shared, AC_HELP_STRING([--enable-shared],
-  [Build shared libraries for Nix [default=yes]]),
-  shared=$enableval, shared=yes)
+  [Build shared libraries for Nix [default=no]]),
```

Comment:
Interesting! @edolstra Two things though:

- In Nixpkgs this will be overriden unless you set ```dontDisableStatic```
- You'll run into a nasty bug where lots of the store implementations aren't included in the final binary because no symbol references them. This is fixed in https://github.com/NixOS/nix/pull/2698 though.